datafusion/substrait/src/logical_plan/consumer.rs

```diff
@@ -934,14 +963,15 @@ pub async fn from_substrait_rel(
                 _ => {
                     not_impl_err!("Unsupported ReadType: {:?}", &read.as_ref().read_type)
                 }
-            }
+            }?;
+            apply_emit_kind(read.common.as_ref(), plan)
```

Comment:
At this time, I've chosen to handle emits on ReadRels as DataFusion Projections.

Future work can be done to further optimize this and push the output mappings further down.

Reply:
same here. I'd leave it as is. pushdown is better done by some sort of substrait optimizer beforehand or by DF optimizer afterwards.